Team Development/server/models/item.js

```diff
@@ -0,0 +1,13 @@
+var mongoose = require('mongoose');
+
+var ItemSchema = new mongoose.Schema ({
+  imagePath: {type: String, required: true},
+  title: {type: String, required: true},
+  description: {type: String, required: true},
+  releaseDate: {type: String, required: true},
+  price: {type: Number, required: true},
+  rating: {type: Number, required: true},
+  reviews: {type: Array, required: true}
```

Comment:
Reviews for relationships should be:
```javascript
reviews: [{type: mongoose.Schemea.Types.ObjectId, ref: 'Review'}]
```

Reply:
Same as above with saveList, reviews and rating shouldnt be required until at instantiation.

Reply:
@TraceLarson yep. For same reasons as saveList, reviews and rating shouldn't be required.

Reply:
An additional note. Item has a OneToMany relationship with ratings. One item can have many ratings.

Reply:
`ratings: [{type: mongoose.Schema.Types.ObjectId, ref: "Rating"}]`

Reply:
We'll have to create a Rating model/schema.

When a user submits an item rating, need to instantiate a Rating which is then associated with both that User and an Item.

When we want to display an item's rating, will need to find that Item, populate its ratings field, and average the value field of all Ratings. This will give an item's rating.

---

Alternatively, we could define three fields on an item, `totalRating`. `numRatings`, and `rating`
`totalRating` : cumulative total of all user ratings for the Item (1-5 for each user rating)
`numRatings` : number of ratings the Item has
`rating` : average rating of the item (calculated by `Total Rating` / `Rating')

Then using a [Pre Hook](https://mongoosejs.com/docs/middleware.html#pre) in the Item schema, we could make sure that every time an Item is saved (would happen when Item is updated to add a new rating to it by incrementing `totalRating` and `numRatings`), it would calculate and set its own `rating` property using its `totalRating` and `numRatings` properties.

Reply:
I actually like the second method better. That way we don't need to mess around with extra schemas, relationships, and documents.

Reply:
Will rating really need its own schema? These are really just placeholders (ratings, reviews). I'm not building those features, so I don't know how they are going to be implemented or how complex they will be.

Shouldn't have been required, though. Went a little heavy on the "command-V". Noted and fixed in next commit.  :)